pipeline/stage02_processing/scripts/spatial_downsampling.py

```diff
@@ -42,48 +36,40 @@ def spatial_smoothing(asig, MACRO_PIXEL_DIM, DIM_X, DIM_Y):
 
     CLI.add_argument("--macro_pixel_dim",  nargs='?', type=int,
                       help="smoothing factor", default=2)
-    CLI.add_argument("--dim_X",  nargs='?', type=int,
-                      help="original x dimension", default=100)
-    CLI.add_argument("--dim_Y",  nargs='?', type=int,
-                      help="original y dimension", default=100)
-    
+
     
     args = CLI.parse_args()
 
     block = load_neo(args.data)
-    asig = block.segments[0].analogsignals[0]
+    block = AnalogSignal2ImageSequence(block)
+    imgseq = block.segments[0].imagesequences[-1]
     
-    MACRO_PIXEL_DIM = args.macro_pixel_dim
-    DIM_X = args.dim_X
-    DIM_Y = args.dim_Y
+    MACRO_PIXEL_DIM = args.macro_pixel_dim    
+    img_reduced = spatial_smoothing(imgseq, MACRO_PIXEL_DIM)
+
+    if args.output_array is not None:
+        np.save(args.output_array, img_reduced[0])
+    if args.output_img is not None:
+        plt.figure()
+        plt.imshow(img_reduced[0])
+        save_plot(args.output_img)
+
+    dim_t = len(img_reduced[:,0,0])
+    dim_x = len(img_reduced[0,:,0])
+    dim_y = len(img_reduced[0,0,:])
+
+    imgseq_reduced = neo.ImageSequence(img_reduced,
+                                   units='dimensionless',
+                                   spatial_scale=imgseq.spatial_scale * MACRO_PIXEL_DIM,
+                                   sampling_rate=imgseq.sampling_rate,
+                                   name='Reduced Images',
+                                   description='Spatial downsampling',
+                                   file_origin=imgseq.file_origin,
+                                   **imgseq.annotations)
+
+    block.segments[0].imagesequences = [imgseq_reduced]
+    block = ImageSequence2AnalogSignal(block)
```

Comment:
In order to replace the analogsignal with the downsampled version you need to clear the analaogsignal list first, (`block.segments[0].analogsignals.clear()`) because `ImageSequence2AnalogSignal()` only appends to that list.